tomcat/src/main/java/com/techcourse/controller/StaticResourceController.java

```diff
@@ -0,0 +1,79 @@
+package com.techcourse.controller;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import org.apache.coyote.http11.HttpHeaders;
+import org.apache.coyote.http11.HttpRequest;
+import org.apache.coyote.http11.HttpResponse;
+import org.apache.coyote.http11.HttpStatus;
+
+public class StaticResourceController extends AbstractController {
+    
+    private static final String STATIC_RESOURCE_PATH = "static";
+    
+    @Override
+    protected HttpResponse doGet(HttpRequest request) throws Exception {
+        return process(request);
+    }
+    
+    @Override
+    protected HttpResponse doPost(HttpRequest request) {
+        return new HttpResponse(
+                "HTTP/1.1",
+                HttpStatus.METHOD_NOT_ALLOWED,
+                HttpHeaders.empty(),
+                ""
+        );
+    }
+    
+    private HttpResponse process(HttpRequest request) throws IOException {
+        String requestUri = request.getRequestUri();
+        String resourcePath = requestUri.equals("/") ? "/index.html" : requestUri;
+        String finalResourcePath = resolveResourcePath(resourcePath);
+        
+        try (InputStream inputStream = StaticResourceController.class.getClassLoader().getResourceAsStream(finalResourcePath)) {
+            if (inputStream == null) {
+                return send404Page();
+            }
+            byte[] responseBody = inputStream.readAllBytes();
+            HttpHeaders headers = HttpHeaders.fromFile(finalResourcePath)
+                .add("Content-Length", String.valueOf(responseBody.length));
```

Comment:
`HttpHeaders.fromFile();` 부분이 무슨 역할을 하는 것인지 메서드 네이밍에 드러나지 않는 것 같습니다.
클래스를 까보고 알게되었는데 해당 파일에 맞게 `Content-Type` 헤더를 추가해주는 메서드네요.
더 직관적인 메서드 네이밍은 없을까요? 이에 대한 모루의 생각이 궁금합니다!

Reply:
추가로 빌더 패턴처럼 `.fromFile()` 의 반환값을 통해 곧바로 `.add()`로 메서드 체이닝을 하고 있는데,
.fromFile()의 네이밍이 와닿지 않아서 메서드 체이닝의 가독성이 좋지 못한 것 같습니다.
만약 둘 다 .add() 메서드로 처리한다면 Headers에 두개의 헤더를 추가한다는 느낌이 명시적으로 들 것 같아요. 모루는 어떻게 생각하시나요?

```suggestion
            HttpHeaders headers = HttpHeaders.
                .add("Content-Type", finalResourcePath)
                .add("Content-Length", String.valueOf(responseBody.length));
```

Reply:
이것도 굳이 fromFile()을 사용하는 것보다 add() 로만 사용하는 것이 나을거 같아요!! 감사해요 히스타~!